tests/test_blocked_well.py

```diff
@@ -0,0 +1,588 @@
+    well_name = 'DOGLEG'
+    bw = resqpy.well.BlockedWell(model,
+                                 well_name = well_name,
+                                 use_face_centres = True,
+                                 grid = grid,
+                                 add_wellspec_properties = True)
+    col_list_orig = ['IW', 'JW', 'L']
+
+    # --------- Act ----------
+    result = bw._BlockedWell__derive_from_wellspec_check_grid_name(check_grid_name = check_grid_name,
+                                                                   grid = grid,
+                                                                   col_list = col_list_orig)
+
+    # --------- Assert ----------
+    assert result[0] == name_for_check
+    assert result[1] == col_list
+
+
+def test_set_for_column(example_model_and_crs):
+
+    # --------- Arrange ----------
+    model, crs = example_model_and_crs
+    grid = RegularGrid(model,
+                       extent_kji = (5, 3, 3),
+                       dxyz = (50.0, -50.0, 50.0),
+                       origin = (0.0, 0.0, 100.0),
+                       crs_uuid = crs.uuid,
+                       set_points_cached = True)
+    grid.write_hdf5()
+    grid.create_xml(write_geometry = True)
+    well_name = 'DOGLEG'
+    bw = resqpy.well.BlockedWell(model, well_name = well_name, use_face_centres = True, add_wellspec_properties = True)
+
+    # --------- Act ----------
+    # populate empty blocked well object for a 'vertical' well in the given column
+    bw.set_for_column(well_name = well_name, grid = grid, col_ji0 = (1, 1))
+
+    # --------- Assert ----------
+    assert bw.cell_count == 5
+    assert bw.node_count == len(
+        bw.node_mds) == len(bw.trajectory.measured_depths) - 1  # tail added to trajectory measured depths
+    assert (len(bw.grid_list) == 1) & (bw.grid_list[0] == grid)
+
+
+def test_derive_from_cell_list(example_model_and_crs):
+
+    # --------- Arrange ----------
+    model, crs = example_model_and_crs
+    grid = RegularGrid(model,
+                       extent_kji = (5, 3, 3),
+                       dxyz = (50.0, -50.0, 50.0),
+                       origin = (0.0, 0.0, 100.0),
+                       crs_uuid = crs.uuid,
+                       set_points_cached = True)
+    grid.write_hdf5()
+    grid.create_xml(write_geometry = True)
+    well_name = 'DOGLEG'
+    cell_kji0_list = np.array([(1, 1, 1), (2, 2, 2), (3, 3, 3), (3, 3, 4)])
+    bw = resqpy.well.BlockedWell(model, well_name = well_name, use_face_centres = True, add_wellspec_properties = True)
+
+    # --------- Act ----------
+    # populate empty blocked well object for a 'vertical' well in the given column
+    bw.derive_from_cell_list(cell_kji0_list = cell_kji0_list, well_name = well_name, grid = grid)
+
+    # --------- Assert ----------
+    # no tail added to trajectory measured depths as well terminates in 4th layer
+    assert bw.node_count == len(bw.node_mds) == len(bw.trajectory.measured_depths)
```

Comment:
Please add an assertion that the blocked well cell count matches the length of the cell_kji0_list that was used to set it up.